content/2018/opening-other-file-formats.md

```diff
@@ -44,30 +44,38 @@ You can find information for writing your own `ftplugin` files at `:h write-file
     endif
     let b:did_ftplugin = 1
 
-to prevent the `ftplugin` from running multiple times on the same buffer. You can then specify commands to be executed on the filetype in question when detected.
+to prevent the `ftplugin` from running multiple times on the same buffer. You can then specify commands to be executed on the filetype in question when detected. If you find this a bit cumbersome, you're not alone. You can bypass the need for such a guard by instead using the `~/.vim/after/ftplugin` directory, which executes after the default `ftplugin` runs (if there is one).
 
 ## Putting it all together
 
 Coming back to our opening statements, how do we make this work for external formats? For starters, make your `ftdetect` file
 
-    :!mkdir -p ~/.vim/ftdetect/pdf.vim
+    :!mkdir -p ~/.vim/ftdetect/video.vim
 
 and edit it with extension detection parameters
 
-    au BufRead,BufNewFile *.pdf set filetype=pdf
+    au BufRead,BufNewFile *.avi,*.mp4,*.mkv,*.mov,*.mpg set filetype=video
 
 then create your `ftplugin` file
 
-    :!mkdir -p ~/.vim/ftplugin/pdf.vim
+    :!mkdir -p ~/.vim/after/ftplugin/video.vim
 
 and edit it appropriately:
 
-    if exists("b:did_ftplugin")
-      finish
-    endif
-    let b:did_ftplugin = 1
-    sil exe "!zathura " . shellescape(expand("%:p")) . " &" | b# | bd# | redraw! | syn on
+    sil exe "!mplayer " . shellescape(expand("%:p")) . " &>/dev/null &" | b# | bd# | redraw! | syn on
```

Comment:
I think using full command names helps readers. `sil` should be `silent`, `b#` sould be `buffer#`, etc.

Also, wouldn't it be easier to use `:help system()`?

    call system('mplayer path/to/foo.mkv')

And maybe let `xdg-open` and the likes deal with choosing the right handler?

Reply:
True, depends on the system you're using. For instance, Mac will use `open`, linux uses `xdg-open` which requires `.desktop` configuration, and windows is a whole different ball game. What would you suggest for cross-platform compatibility?

Reply:
I use [this snippet](https://github.com/romainl/vim-devdocs/blob/master/plugin/devdocs.vim#L16)